okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.java

```diff
@@ -131,4 +145,28 @@ public CallServerInterceptor(boolean forWebSocket) {
 
     return response;
   }
+
+  private long headerLength(Headers headers) {
+    long length = 0;
+
+    for (int i = 0, size = headers.size(); i < size; i++) {
+      length += headers.name(i).length();
+      length += headers.value(i).length();
+    }
+
+    return length;
+  }
+
+  final class CountingSink extends ForwardingSink {
```

Comment:
nice

Reply:
Maybe it should be `static final class`?

Reply:
Thanks @PromanSEW https://github.com/square/okhttp/pull/3544